### PR TITLE
Remove Rubygem#to_param

### PR DESCRIPTION
**Pull request by segiddins**
Instead, manually pass slug to path helpers, similar to what we do for versions

Closes https://github.com/rubygems/rubygems.org/issues/3984

### CODE REVIEW

**Comment by simi** [comment]
I did something similar at https://github.com/rubygems/rubygems.org/commit/7959c5eef3705f0e13e303184085e3a1bfac0b32. Feel free to inspire if there will be any problem on this PR. It seems mostly the same.

File: test/integration/avo/rubygems_test.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Avo::RubygemsTest < ActionDispatch::IntegrationTest
     assert_response :success
     assert page.has_content? rubygem.name
 
-    get avo.resources_rubygem_path(rubygem)
+    get avo.resources_rubygem_path(rubygem.slug)
```
Comment (simi) [inline test/integration/avo/rubygems_test.rb:20]: If I understand it well, this is not needed, since in admin we use `id`.

Comment (segiddins) [reply test/integration/avo/rubygems_test.rb:20]:
```suggestion
    get avo.resources_rubygem_path(rubygem)
```

**Review by simi — APPROVED** [review]
Looks good to me. I wasn't sure this kind of relatively long patch would be accepted, but it looks there is no other simple way to fix the admin problem.

**Review by martinemde — APPROVED** [review]
I don't quite understand the benefit of this, but it looks fine given a scan of the changes.

**Comment by simi** [comment]
> I don't quite understand the benefit of this, but it looks fine given a scan of the changes.

@martinemde Avo admin part is using `to_param` to craft URLs as well, but since we can't specify admin routes manually, it recognizes url like `/active_model_serializers_rails_2.3` as having id of `active_model_serializers_rails_2` and `3` as extension. https://github.com/rubygems/rubygems.org/issues/3984

Removing `to_param` uses default implementation using `id` and thus admin routes are again `/123` instead of gem slug. To keep previous app urls same as before (using gem slug, not id), we need to pass slug directly to path helpers.

**Pull request merged**